exciting_environments/core_env.py

```diff
@@ -149,109 +143,101 @@ def _test_rew_func(self, func):
             compatible(bool): Environment compatibility.
         """
         try:
-            out = func(
-                jnp.zeros([self.batch_size, int(len(self.obs_description))]))
+            out = jax.vmap(func, in_axes=(0, 0, self.in_axes_env_properties.action_constraints))(
+                jnp.zeros((self.batch_size, int(len(self.obs_description)))),
+                jnp.zeros((self.batch_size, int(
+                    len(list(vars(self.env_properties.action_constraints).values()))))),
+                self.env_properties.action_constraints
+            )
         except:
             raise Exception(
-                "Reward function should be using obs matrix as only parameter")
+                "Reward function should be using observation matrix, action and self.env_properties.action_constraints as parameters")
```

Comment:
I think this error message could be a bit confusing. There are certainly cases where the function call breaks and the reason is something other than the input arguments. Instead of doing a try -> except statement here, you should just test a couple of different inputs and the output shapes here.

I am also unsure if we need to provide a test for the reward function. We should present a e.g. a header or documentation.